packages/loader/container-loader/src/container.ts

```diff
@@ -225,7 +225,14 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     private storageService: IDocumentStorageService | undefined | null;
     private blobsCacheStorageService: IDocumentStorageService | undefined;
 
-    private _clientId: string | undefined;
+    /**
+     * Our current (and previous) clientIds, with the latest (and current) at the
+     * end of the array.  We keep a history so that messages with different IDs
+     * can be identified as originating from this client in cases of multiple
+     * disconnects/reconnects.
+     */
+    private readonly _clientIds: string[] = [];
```

Comment:
we should add a new array, rather than changing this variable

Reply:
this will keep the change as minimal as possible

Reply:
I reverted the change to the check in `protocol.quorum.on("removeMember"` and added a logging check in processSignal as well.  So functionally the only change now should be processing messages/signals as local (when the matched clientId is further back in the history) and the telemetry.